scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,327 @@
+/*!
+Copyright 2018 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+// This script modifies the preferences data base:
+// 1. Finds all the Prefs Safes of type "snapset" (prefsSafesType = "snapset"),
+// 2. Finds all the GPII Keys associated with each snapset Prefs Safe
+// 3. Deletes the found Prefs Safes and associated GPII Keys
+//
+// A sample command that runs this script:
+// node deleteSnapsets.js $COUCHDBURL
+
+"use strict";
+
+var http = require("http"),
+    url = require("url"),
+    fs = require("fs"),
+    fluid = require("infusion");
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.dataLoader");
+fluid.setLogging(fluid.logLevel.INFO);
+
+var dbLoader = gpii.dataLoader;
+
+// Handle command line
+if (process.argv.length < 5) {
+    fluid.log("Usage: node deleteAndLoadSnapsets.js $COUCHDB_URL $STATIC_DATA_DIR $BUILD_DATA_DIR $BUILD_DEMOUSER_DIR [--justDelete]");
+    process.exit(1);
+}
+dbLoader.couchDbUrl = process.argv[2];
```

Comment:
Stashing these variable command-line arguments onto the global namespace and then fishing them out in various functions seems like an unfortunate design. Please write a standard top-level function which accumulates these into a local variable rather than writing this material at top level, and then feed these in as function arguments.

Reply:
Okay.